mirascope/core/base/_utils.py

```diff
@@ -136,13 +147,16 @@ def convert_function_to_base_tool(
             doesn't have a docstring description.
     """
     docstring = None
-    if fn.__doc__:
-        docstring = parse(fn.__doc__)
+    func_doc = __doc__ or fn.__doc__
+    if func_doc:
+        docstring = parse(func_doc)
 
     field_definitions = {}
     hints = get_type_hints(fn)
+    has_self_or_cls = False
     for i, parameter in enumerate(inspect.signature(fn).parameters.values()):
         if parameter.name == "self" or parameter.name == "cls":
+            has_self_or_cls = True
```

Comment:
what if instead we did something like this?

```python
if (has_self := (param.name == "self")) or parameter.name == "cls":
    continue

...
```

Then we can use just `has_self` separately since we aren't including the `cls` below?

Reply:
It is ok to separate them into `has_self`.

However, in that example, the value is assigned to `has_self` in every loop, so if there is more than one parameter, `has_self` will certainly be `False`.

Reply:
Oh good call you are right.